hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonRecordReader.java

```diff
@@ -116,6 +116,20 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext context)
     return readSupport.readRow(carbonIterator.next());
   }
 
+  /**
+   * get batch result
+   *
+   * @param batch batch size
+   * @return rows
+   */
+  public List<Object[]> getBatchValue(int batch) {
+    rowCount += batch;
```

Comment:
we may not get complete batch size data from RowBatch.nextBatch(), it can return lesser data also. So, incrementing rowcount as batchsize is wrong. 
we need to increment this value same as returned rows from batch iterator.

Reply:
ok, optimized